pyTigerGraph/pyTigerGraphBase.py

```diff
@@ -188,6 +183,23 @@ def __init__(self, host: str = "http://127.0.0.1", graphname: str = "MyGraph",
             self.gsUrl = self.host + ":" + self.gsPort
         self.url = ""
 
+        if self.username.startswith("arn:aws:iam::"):
+            import boto3
+            from botocore.awsrequest import AWSRequest
+            from botocore.auth import SigV4Auth
+            # Prepare a GetCallerIdentity request.
+            request = AWSRequest(
+                method="POST",
+                url="https://sts.amazonaws.com/?Action=GetCallerIdentity&Version=2011-06-15",
+                headers={
+                    'Host': 'sts.amazonaws.com'
+                })
+            # Get headers
+            SigV4Auth(boto3.Session().get_credentials(), "sts", "us-east-1").add_auth(request)
```

Comment:
Just to confirm, `'sts'` and `'us-east-1'` will always be valid static params? What happens if the us east region is down?

Reply:
Good question. "sts" is the service to use for auth and it is always valid. "us-east-1" is used in generating the signature. pyTigerGraph never sends the request out. It sends the signature to the gsql server and gsql server handles the authentication with aws iam. Based on AWS doc, us-east-1 should always work unless there is an outage at AWS https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html: 
"By default, AWS Security Token Service (AWS STS) is available as a global service, and all AWS STS requests go to a single endpoint at https://sts.amazonaws.com. Global requests map to the US East (N. Virginia) Region. AWS recommends using Regional AWS STS endpoints instead of the global endpoint to reduce latency"